scripts/globals/abyssea.lua

```diff
@@ -483,102 +474,83 @@ function AddPlayerLights(player, light, amount)
     -------------------------------------------------------------------------------------
     -- lights: 1: pearl, 2: azure, 3: ruby, 4: amber, 5: gold, 6: silver, 7: ebon
     -------------------------------------------------------------------------------------
+
+    -------------------------------------------------------------------------------------
+    -- Pearlescent Light
+    -------------------------------------------------------------------------------------
     if light == 1 then
+        local pearl = player:getCharVar("pearlLight")
         if pearl + lightamount > 230 then
             player:setCharVar("pearlLight",230)
         else
             player:setCharVar("pearlLight",pearl + lightamount)
         end
-
         player:messageSpecial(ID.text.BODY_EMITS_PEARL_LIGHT,tiermsg)
+    -------------------------------------------------------------------------------------
+    -- Azure Light
+    -------------------------------------------------------------------------------------
     elseif light == 2 then
+        local azure = player:getCharVar("azureLight")
         if azure + lightamount > 255 then
             player:setCharVar("azureLight",255)
         else
             player:setCharVar("azureLight",azure + lightamount)
         end
-
         player:messageSpecial(ID.text.BODY_EMITS_AZURE_LIGHT,tiermsg)
+    -------------------------------------------------------------------------------------
+    -- Ruby Light
+    -------------------------------------------------------------------------------------
     elseif light == 3 then
+        local ruby = player:getCharVar("rubyLight")
         if ruby + lightamount > 255 then
             player:setCharVar("rubyLight",255)
         else
             player:setCharVar("rubyLight",ruby + lightamount)
         end
-
         player:messageSpecial(ID.text.BODY_EMITS_RUBY_LIGHT,tiermsg)
+    -------------------------------------------------------------------------------------
+    -- Amber Light
+    -------------------------------------------------------------------------------------
     elseif light == 4 then
+        local amber = player:getCharVar("amberLight")
         if amber + lightamount > 255 then
             player:setCharVar("amberLight",255)
         else
             player:setCharVar("amberLight",amber + lightamount)
         end
-
         player:messageSpecial(ID.text.BODY_EMITS_AMBER_LIGHT,tiermsg)
+    -------------------------------------------------------------------------------------
+    -- Golden Light
+    -------------------------------------------------------------------------------------
     elseif light == 5 then
```

Comment:
```lua

    -- Golden Light
    elseif light == 5 then
```
instead of brick houses for comments

Reply:
or enums /shrug

Reply:
I haven't looked at any code, just checking notifications:

Enums~! No magic numbers...!

Reply:
All magic number should have been replaced with enums now

Reply:
can mark this convo resolved since it has enums now